### PR TITLE
zephyr/CMakeLists.txt: remove '..' in include paths 

### PR DESCRIPTION
**Pull request by marc-hb**
**zephyr/CMakeLists.txt: remove '..' in include paths**

CMake seems to behave differently on Linux and Windows: it generates
different `-I` command line parameters. This results in spurious
`__FILE__` mismatches and non-reproducible builds when using
CONFIG_ASSERT, see example in https://github.com/thesofproject/sof/pull/7428.

On Windows, '..' seem resolved more often which also seems to convert
forward slashes to backslashes.

They are also less readable and wasting a bit of space. Remove them
using cmake_path(SET ...)

---------

Zephyr's `-fmacro-prefix-map` is working and keeps the builds reproducible when using a recent enough toolchain.

As found in the CONFIG_ASSERT PR https://github.com/thesofproject/sof/pull/6530#issuecomment-1482330214 this is not true for old Xtensa toolchains which don't support `-fmacro-prefix-map`

### CODE REVIEW

**Comment by marc-hb** [comment]
Some zephyr.lst differences in https://github.com/thesofproject/sof/actions/runs/4662244951/jobs/8252512487?pr=7428, interesting...

**Comment by marc-hb** [comment]
The build on Windows has a mix of forward and backward slashes. There are 63 WEST_TOPDIR paths with forward slashes and 6 with backward slashes:
```
strings build-sof-staging/sof-info/mtl/stripped-zephyr.elf | grep TOPDIR

...
WEST_TOPDIR/sof/src/library_manager/lib_notification.c
WEST_TOPDIR/sof/src/samples/audio/detect_test.c
WEST_TOPDIR/sof/src/schedule/zephyr_ll.c
WEST_TOPDIR/sof/zephyr/include/rtos/task.h
WEST_TOPDIR/sof/zephyr/lib/cpu.c
WEST_TOPDIR/sof/zephyr/wrapper.c
WEST_TOPDIR/zephyr/arch/common/sw_isr_common.c
WEST_TOPDIR/zephyr/arch/xtensa/core/xtensa-asm2.c
WEST_TOPDIR/zephyr/arch/xtensa/include/kernel_arch_func.h
WEST_TOPDIR/zephyr/drivers/dai/intel/dmic/dmic.c

...
WEST_TOPDIR/zephyr/subsys/logging/log_output.c
WEST_TOPDIR/zephyr/subsys/pm/device_runtime.c
WEST_TOPDIR/zephyr/subsys/pm/pm.c
WEST_TOPDIR/zephyr/subsys/pm/policy.c
WEST_TOPDIR\sof\src\include\sof\audio\audio_stream.h
WEST_TOPDIR\sof\src\include\sof\audio\component_ext.h
WEST_TOPDIR\sof\src\include\sof\coherent.h
WEST_TOPDIR\sof\src\include\sof\schedule\ll_schedule_domain.h
WEST_TOPDIR\sof\src\platform\intel\ace\include\ace\lib\mailbox.h
WEST_TOPDIR\sof\xtos\include\sof\lib\mailbox.h
```

**Review by kv2019i — COMMENTED** [review]
Looks good. One question , see inline comment

File: .github/workflows/zephyr.yml [hunk 1]
```diff
@@ -131,7 +131,7 @@ jobs:
       - name: build
         run: cd workspace && ./sof/zephyr/docker-run.sh
              ./sof/zephyr/docker-build.sh --cmake-args=-DEXTRA_CFLAGS=-Werror
-             --cmake-args=--warn-uninitialized ${{ matrix.IPC_platforms }}
+             --cmake-args=--warn-uninitialized -d ${{ matrix.IPC_platforms }}
```
[the remark this answers is not in the document]
Comment (aborisovich) [reply .github/workflows/zephyr.yml:134]: I agree with @kv2019i on this one. As you said Marc, building whole array with -d flag would result in so many builds...
I think we should stick with just a "production" build

**Review by aborisovich — APPROVED** [review]

**Comment by marc-hb** [comment]
@kv2019i , @aborisovich I removed the github checks changes for now, let's discuss them later in another PR. It's more urgent to *fix* CONFIG_ASSERT reproducibility than to *test* it. So this PR should be a no-brainer now.

**Review by kv2019i — APPROVED** [review]

File: .github/workflows/zephyr.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marc-hb) [reply .github/workflows/zephyr.yml]: Github Actions allow "sparse" matrices. So I found a way to add `-d` only for MTL, submitted in
- #7508

**Pull request merged**